### PR TITLE
[zh-tw] Deep learning

### PR DESCRIPTION
**Pull request by kevingo**
This PR contains two parts:

1. Add zh-tw folder and raw markdown.
2. Add deep learning zh-tw translation.

### CODE REVIEW

**Comment by shervinea** [comment]
Thanks for your great work @kevingo!

A few comments. Could you please:
- pull the latest version of the master branch? There seems to be a few missing questions to translate at the end of each cheatsheet.
- do this PR only by adding the deep learning cheatsheet? 
- Make sure that your name in the `CONTRIBUTORS` file is in the format `Firstname Lastname`?

**Comment by kevingo** [comment]
HI @shervinea , I've modified the PR as you suggested. Please check again. Thank you.

And I just want to leave the `CONTRIBUTORS` name with `kevingo` only, is it ok ?

**Comment by shervinea** [comment]
@kevingo thanks for getting back quickly! It seems that questions 50-54 are still missing (cf. [this template](https://github.com/shervinea/cheatsheet-translation/blob/master/es/cheatsheet-deep-learning.md)).

Sure, we can leave it as `kevingo` if it is what works best.

**Comment by kevingo** [comment]
@shervinea Thanks for remind. I've already update the missing parts (50-54).

**Comment by shervinea** [comment]
Thanks @kevingo! Now, let's wait for a reviewer to go through the translation.

File: zh-tw/cheatsheet-deep-learning.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+6. **By noting i the ith layer of the network and j the jth hidden unit of the layer, we have:**
+
+&#10230;
+我們使用 i 來代表網路的第 i 層、j 來代表某一層中第 j 個隱藏神經元的話，我們可以得到下面得等式：
+<br>
+
+7. **where we note w, b, z the weight, bias and output respectively.**
+
+&#10230;
+其中，我們分別使用 w 來代表權重、b 代表偏差、z 代表輸出。
```
Comment (TobyOoO) [inline zh-tw/cheatsheet-deep-learning.md]: > 其中，我們分別使用 w 來代表權重、b 代表偏差項、z 代表輸出的結果。

File: zh-tw/cheatsheet-deep-learning.md [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+<br>
+
+8. **Activation function ― Activation functions are used at the end of a hidden unit to introduce non-linear complexities to the model. Here are the most common ones:**
+
+&#10230;
+激勵函數 - 激勵函數是為了在神經元中帶入非線性的轉換而設計。底下是一些常見函數：
```
Comment (TobyOoO) [inline zh-tw/cheatsheet-deep-learning.md]: > Activation function - Activation function 是為了在每一層尾端的神經元帶入非線性轉換而設計的。底下是一些常見 Activation function ： 

建議 Activation function 保留原文，因為激勵函數可能會跟 Reward function 混淆。

File: zh-tw/cheatsheet-deep-learning.md [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+10. **Cross-entropy loss ― In the context of neural networks, the cross-entropy loss L(z,y) is commonly used and is defined as follows:**
+
+&#10230;
+交叉熵損失函式
+<br>
+
+11. **Learning rate ― The learning rate, often noted α or sometimes η, indicates at which pace the weights get updated. This can be fixed or adaptively changed. The current most popular method is called Adam, which is a method that adapts the learning rate.**
+
+&#10230;
+學習速率 - 學習速率通常用 α 或 η 來表示，目的是用來控制權重更新的速度。學習速度可以是一個固定值，或是隨著訓練的過程改變。現在最熱門的方法叫做 Adam，是一種隨著訓練過程改變的學習速度。
```
Comment (TobyOoO) [inline zh-tw/cheatsheet-deep-learning.md]: > 學習速率 - 學習速率通常用 α 或 η 來表示，目的是用來控制權重更新的速度。學習速度可以是一個固定值，或是隨著訓練的過程改變。現在最熱門的最佳化方法叫作 Adam，是一種隨著訓練過程改變學習速率的最佳化方法。

File: zh-tw/cheatsheet-deep-learning.md [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+20. **Convolutional Neural Networks**
+
+&#10230;
+卷積神經網絡
+<br>
+
+21. **Convolutional layer requirement ― By noting W the input volume size, F the size of the convolutional layer neurons, P the amount of zero padding, then the number of neurons N that fit in a given volume is such that:**
+
+&#10230;
+卷積層的需求 - 我們使用 W 來表示輸入的尺寸、F 代表卷積層的 filter 尺寸、P 代表使用的 pad 數量，S 代表 stride 的數量，則輸出的尺寸可以透過以下的公式表示：
```
Comment (TobyOoO) [inline zh-tw/cheatsheet-deep-learning.md]: > 卷積層的需求 - 我們使用 W 來表示輸入資料的維度大小、F 代表卷積層的 filter 尺寸、P 代表對資料墊零 (zero padding) 使資料長度齊一後的長度，S 代表卷積後取出的特徵 stride 數量，則輸出的維度大小可以透過以下的公式表示：

File: zh-tw/cheatsheet-deep-learning.md [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+22. **Batch normalization ― It is a step of hyperparameter γ,β that normalizes the batch {xi}. By noting μB,σ2B the mean and variance of that we want to correct to the batch, it is done as follows:**
+
+&#10230;
+批次正規化 (Batch normalization) - 它是一個藉由 γ,β 兩個超參數來正規化每個批次 {xi} 的過程。每一次正規化的過程，我們使用 μB,σ2B 分別代表平均數和變異數。請參考以下公式：
+<br>
+
+23. **It is usually done after a fully connected/convolutional layer and before a non-linearity layer and aims at allowing higher learning rates and reducing the strong dependence on initialization.**
+
+&#10230;
+批次正規化的動作通常在全連接層/卷積層之後、在非線性層之前進行。目的在於容許更高的學習速率，並且減少對於初始化資料的依賴
```
Comment (TobyOoO) [inline zh-tw/cheatsheet-deep-learning.md]: > 批次正規化的動作通常在全連接層/卷積層之後、在非線性層之前進行。目的在於接納更高的學習速率，並且減少該批次學習初期對取樣資料特徵的依賴性。

File: zh-tw/cheatsheet-deep-learning.md [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+29. **Reinforcement Learning and Control**
+
+&#10230;
+強化學習及控制
+<br>
+
+30. **The goal of reinforcement learning is for an agent to learn how to evolve in an environment.**
+
+&#10230;
+強化學習的目標就是為了讓代理 (agent) 能夠學習在環境中進化
```
Comment (TobyOoO) [inline zh-tw/cheatsheet-deep-learning.md]: >  強化學習的目標就是為了讓代理者 (agent) 能夠在環境中自我學習、自我演化。

File: zh-tw/cheatsheet-deep-learning.md [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+48. **times took action a in state s**
+
+&#10230;
+從狀態 s 所採取行為的次數
+<br>
+
+49. **Q-learning ― Q-learning is a model-free estimation of Q, which is done as follows:**
+
+&#10230;
+Q-learning 演算法 - Q-learning 演算法是針對 Q 的一個 model-free 的估計，如下：
```
Comment (TobyOoO) [inline zh-tw/cheatsheet-deep-learning.md]: > Q-learning 演算法 - Q-learning 演算法是一個不需要模型，即可求出 Q 估計值的一個演算法，作法如下：

**Comment by kevingo** [comment]
Thanks for the reviewing by @TobyOoO . All suggestions are professional. I will adjust as recommended.

**Comment by kevingo** [comment]
@shervinea @TobyOoO I've modified the content accordingly. Thanks!

**Comment by shervinea** [comment]
Thanks @kevingo @TobyOoO. Are we good to go?

Also, @TobyOoO could you please add your name to the contributors' list?

**Comment by kevingo** [comment]
Hi @shervinea, I've check with @TobyOoO and help him to add his name to `CONTRIBUTOR`. Please help to check. Thanks 👍

**Comment by shervinea** [comment]
Thank you very much @kevingo and @TobyOoO! Merging it right now.

**Pull request merged**